powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/repository/ActivationHistoryRepository.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * PowerAuth Server and related software components
+ * Copyright (C) 2017 Lime - HighTech Solutions s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.getlime.security.powerauth.app.server.database.repository;
+
+import io.getlime.security.powerauth.app.server.database.model.entity.ActivationHistoryEntity;
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Database repository for accessing activation history.
+ *
+ * @author Roman Strobl, roman.strobl@lime-company.eu
+ */
+public interface ActivationHistoryRepository extends CrudRepository<ActivationHistoryEntity, Long> {
+
+    List<ActivationHistoryEntity> findByActivation_ActivationIdAndTimestampCreatedBetweenOrderByTimestampCreatedDesc(String activationId, Date startingDate, Date endingDate);
```

Comment:
Please use HQL / `@Query` annotation, so that you do not write the code the way I did - the method name is way too long...